x-pack/plugins/siem/public/components/super_date_picker/index.test.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mount } from 'enzyme';
+import * as React from 'react';
+
+import { mockGlobalState, TestProviders } from '../../mock';
+import { createStore, State } from '../../store';
+
+import { SuperDatePicker } from '.';
+
+describe('SIEM Super Date Picker', () => {
+  describe('#SuperDatePicker', () => {
+    const state: State = mockGlobalState;
+    let store = createStore(state);
+
+    beforeEach(() => {
+      store = createStore(state);
+    });
+
+    test('Can select Today (it is a relative date)', () => {
+      const wrapper = mount(
+        <TestProviders store={store}>
+          <SuperDatePicker id="global" />
+        </TestProviders>
+      );
+      wrapper
+        .find('[data-test-subj="superDatePickerToggleQuickMenuButton"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
+        .find('[data-test-subj="superDatePickerCommonlyUsed_Today"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      expect(store.getState().inputs.global.timerange.kind).toBe('relative');
+      expect(store.getState().inputs.global.timerange.option).toBe('now/d');
+    });
+
+    test('Recently used date ranges', () => {
+      const wrapper = mount(
+        <TestProviders store={store}>
+          <SuperDatePicker id="global" />
+        </TestProviders>
+      );
+      wrapper
+        .find('[data-test-subj="superDatePickerToggleQuickMenuButton"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
+        .find('button.euiQuickSelect__applyButton')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
+        .find('[data-test-subj="superDatePickerToggleQuickMenuButton"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
+        .find('[data-test-subj="superDatePickerCommonlyUsed_Today"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
+        .find('[data-test-subj="superDatePickerToggleQuickMenuButton"]')
+        .first()
+        .simulate('click');
+      wrapper.update();
+
+      wrapper
```

Comment:
It appears that the `Today` button is clicked on [line 71](https://github.com/elastic/kibana/pull/34806/files#diff-cfffba54eebe4d47bd495ad850122aacR71):

>       wrapper
>        .find('[data-test-subj="superDatePickerCommonlyUsed_Today"]')
>        .first()
>        .simulate('click');
>      wrapper.update();

and now the `Year to date` button is being clicked here, so this might be another instance to consider refactoring each button click into a separate self-contained test that clicks the button, and then `expects` a result. (If so, this refactoring would apply to the rest of the actions in this test.)

Reply:
here it was just to make sure that we do not add any duplicate in the recently used view

Reply:
> here it was just to make sure that we do not add any duplicate in the recently used view

Is it possible to prevent duplicates another way, for example by, initializing the state of the store to ensure there are no duplicates, or via multiple, smaller tests? If not, consider adding a comment that explains the purpose of these clicks and the desired effects (e.g. preventing duplicates).

Reply:
As discussed, I will clean up the test to be more readable